fili-core/src/main/java/com/yahoo/bard/webservice/druid/util/ThetaSketchFieldConverter.java

```diff
@@ -30,21 +31,45 @@ public  ThetaSketchAggregation asInnerSketch(SketchAggregation candidate) {
         return new ThetaSketchAggregation(candidate.getName(), candidate.getFieldName(), candidate.getSize());
     }
 
-   @Override
-    public ThetaSketchEstimatePostAggregation asSketchEstimate(SketchAggregation candidate) {
+    /**
+     * Get the candidate Aggregation as a SketchEstimatePostAggregation.
+     *
+     * @param candidate  Aggregation to "convert"
+     *
+     * @return The Aggregation as a SketchEstimatePostAggregation
+     */
+    public ThetaSketchEstimatePostAggregation asSketchEstimate(Aggregation candidate) {
         String name = candidate.getName() + "_estimate";
         PostAggregation field = new FieldAccessorPostAggregation(candidate);
         return new ThetaSketchEstimatePostAggregation(name, field);
     }
 
+    /**
+     * Get the PostAggregation as a SketchEstimatePostAggregation.
+     *
+     * @param postAggregation  PostAggregation to "convert"
+     *
+     * @return The PostAggregation as a SketchEstimatePostAggregation
+     */
+    public ThetaSketchEstimatePostAggregation asSketchEstimate(PostAggregation postAggregation) {
+        String name = postAggregation.getName() + "_estimate";
+        return new ThetaSketchEstimatePostAggregation(name, postAggregation);
+    }
+
     @Override
     public FuzzySetPostAggregation asSketchEstimate(MetricField field) {
         if (field.isSketch()) {
```

Comment:
This is my personal preference and you can decide whether to take it or not:

I prefer reverting the condition like this

```java
if (!field.isSketch()) {
    String message = "Given metric field " + field.toString() + " isn't sketch";
    LOG.error(message);
    throw new IllegalArgumentException(message);
}

if (field instanceof PostAggregation) {
    return asSketchEstimate((PostAggregation) field);
}
else if (field instanceof Aggregation) {
    return asSketchEstimate((Aggregation) field);
} else {
    String message = "Given metric field " + field.toString() + " is neither a type of " +
            "PostAggregation nor Aggregation";
    LOG.error(message);
    throw new IllegalArgumentException(message);
}
```

This moves more complicated logic out of if branch, making the code easier to read.